surefire-providers/common-junit48/src/main/java/org/apache/maven/surefire/common/junit48/FailingMethodFilter.java

```diff
@@ -1,85 +0,0 @@
-package org.apache.maven.surefire.common.junit48;
-
-/*
- * Licensed to the Apache Software Foundation (ASF) under one
- * or more contributor license agreements.  See the NOTICE file
- * distributed with this work for additional information
- * regarding copyright ownership.  The ASF licenses this file
- * to you under the Apache License, Version 2.0 (the
- * "License"); you may not use this file except in compliance
- * with the License.  You may obtain a copy of the License at
- *
- *     http://www.apache.org/licenses/LICENSE-2.0
- *
- * Unless required by applicable law or agreed to in writing,
- * software distributed under the License is distributed on an
- * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
- * KIND, either express or implied.  See the License for the
- * specific language governing permissions and limitations
- * under the License.
- */
-
-import org.apache.maven.shared.utils.io.SelectorUtils;
-import org.junit.runner.Description;
-import org.junit.runner.manipulation.Filter;
-
-import java.util.Map;
-import java.util.Set;
-
-/**
- * Only run test methods in the given input map, indexed by test class
- */
-final class FailingMethodFilter
```

Comment:
I don't think this can be removed. I have to check the algorithms of FailingMethodFilter and MatchMethodDescriptions.

Reply:
Before deleting a class, tests with high coverage should be added and new implementation should pass on them.

Reply:
This is not really possible as as the way we're dealing selecting the tests to rerun has changed fundamentally. I have however added `JUnit4ProviderUtilTest#testGenerateFailingTestDescriptions` to show equivalence to `JUnit4ProviderUtilTest#testGenerateFailingTests`

This shows that the set of failed tests is the same as before - though described in a different format.